plugins/cookie_manager/lib/src/cookie_mgr.dart

```diff
@@ -123,7 +123,9 @@ class CookieManager extends Interceptor {
     // users will be available to handle cookies themselves.
     final isRedirectRequest = statusCode >= 300 && statusCode < 400;
     // Saving cookies for the original site.
-    await cookieJar.saveFromResponse(response.realUri, cookies);
+    final originalUri = response.requestOptions.uri;
+    final realUri = originalUri.resolveUri(response.realUri);
```

Comment:
Do we have any standards that indicate we should resolve the real dest based on the original one?

Reply:
https://www.rfc-editor.org/rfc/rfc7231#section-7.1.2

> The field value consists of a single URI-reference.  When it has the
   form of a relative reference ([RFC3986], Section 4.2), the final
   value is computed by resolving it against the effective request URI
   ([RFC3986], Section 5).

Reply:
Great. Could you add it as a comment, so we can track it back somedays if we forgot it.

Reply:
@s0nerik